mantidproject/mantid
# Use setuptools develop for PythonInterface target

**Description of work.**
Stops copying of the PythonInterface python code for developer builds. Uses `setuptools.develop` to create an egg-link instead. The `.pyd` files are now inside the main build bin dir: `bin/CONFIG/*.pyd`. They are still copied for packaged builds.

Added `mantidbuild.py.in` file configured with the current build directory - this is used to correctly load `Mantid.properties` for tests. I'm not sure if this is the best way to do it, but I couldn't think of another way to find the build directory for out-of-source builds.

Extended `add_python_package` to support a `EGGLINKNAME` which is used as the name for the egg-link (PythonInterface's egg-link is called `mantid.egg-link`).

Adds a check for a `mantid` folder present in the bin directory. If so it will recursively delete it, as it is no longer necessary.

Changed imports to be absolute in most places in `PythonInterface`.

**To test:**
Builds should pass.
Locally try running MantidPlot and Workbench after a full build, all should run as expected. Try running some Python tests (it's a special case for importing `Mantid.properties`)

Fixes #25139

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
